### ISSUE TITLE
PCL CUDA build issues

### ISSUE THREAD

**Issue by chrispduck** [PointCloudLibrary/pcl#3650]
Is there an up-to-date PCL CUDA/GPU installation guide? I am getting several CUDA errors when building PCL CUDA with CUDA 9.0. These issues were outline in the following post but no solution is working for me. I've tried to compile both PCL 1.9 and PCL 1.10 with CUDA 9.0 and am getting errors the same as [here](https://github.com/PointCloudLibrary/pcl/issues/2541). Pasted below for reference:

> 
> I seem to have a related compilation error with the same build.
> This has been a major bottleneck in my project, hence any help is greatly appreciated!
> 
> - Operating System and version: Ubuntu 16.04
> - CUDA: 9.0
> - PCL Version: 1.9.1
> 
> ```
> [ 44%] Building NVCC (Device) object gpu/utils/CMakeFiles/pcl_gpu_utils.dir/src/pcl_gpu_utils_generated_repacks.cu.o
> /home/funaizhang/Downloads/pcl/gpu/utils/src/repacks.cu(84): error: identifier "copy_fields_t" is undefined
> 
> /home/funaizhang/Downloads/pcl/gpu/utils/src/repacks.cu(84): error: expected a ";"
> 
> /home/funaizhang/Downloads/pcl/gpu/utils/src/repacks.cu(96): error: namespace "pcl::device" has no member "copy_fields_t"
> 
> 3 errors detected in the compilation of "/tmp/tmpxft_0000124e_00000000-13_repacks.compute_70.cpp1.ii".
> CMake Error at pcl_gpu_utils_generated_repacks.cu.o.cmake:268 (message):
>   Error generating file
>   /home/funaizhang/Downloads/pcl/release/gpu/utils/CMakeFiles/pcl_gpu_utils.dir/src/./pcl_gpu_utils_generated_repacks.cu.o
> 
> 
> gpu/utils/CMakeFiles/pcl_gpu_utils.dir/build.make:241: recipe for target 'gpu/utils/CMakeFiles/pcl_gpu_utils.dir/src/pcl_gpu_utils_generated_repacks.cu.o' failed
> make[2]: *** [gpu/utils/CMakeFiles/pcl_gpu_utils.dir/src/pcl_gpu_utils_generated_repacks.cu.o] Error 1
> CMakeFiles/Makefile2:1283: recipe for target 'gpu/utils/CMakeFiles/pcl_gpu_utils.dir/all' failed
> make[1]: *** [gpu/utils/CMakeFiles/pcl_gpu_utils.dir/all] Error 2
> Makefile:149: recipe for target 'all' failed
> make: *** [all] Error 2
> ```
> 
> _Originally posted by @Funaizhang in https://github.com/PointCloudLibrary/pcl/issues/2541#issuecomment-516837909_
> 

To summarise:
What are the current guidelines for installing PCL with GPU/CUDA support for UBUNTU 16.04? CUDA 10.0, CUDA 9.0 or older? 
Are there any guides for implementing PCL GPU C++ code once installed? I am struggling to find documentation and examples. I would like to have GPU support for ICP registration, and for other segmentation algortihms. 

Any help would be hugely appreciated.

**Comment by hyc-xyz** [comment]
I have the same question here

**Comment by kunaltyagi** [comment]
I'm not an active user of PCL cuda, but I use latest GCC for source, latest (nvcc + most recent compatible gcc) for device files compilation

**Comment by kunaltyagi** [comment]
AFAIK, CUDA 9 isn't compatible with PCL with default dependencies. We're still figuring out the minimum versions and an updated Dockerfile will be available soon thanks to @haritha-j and @shrijitsingh99

**Comment by kunaltyagi** [comment]
CUDA is now built on the CI. Thanks to @shrijitsingh99, you can find the configuration in the [Ubuntu 18.04 Clang CI](https://github.com/PointCloudLibrary/pcl/blob/master/.ci/azure-pipelines/build/ubuntu.yaml). Please refer to Ubuntu Dockerfile for dependencies
